server/src/main/java/org/elasticsearch/repositories/ShardGenerations.java

```diff
@@ -219,12 +220,23 @@ public Builder put(IndexId indexId, int shardId, SnapshotsInProgress.ShardSnapsh
         }
 
         public Builder put(IndexId indexId, int shardId, ShardGeneration generation) {
+            assert noDuplicateIndicesWithSameName(indexId)
+                : Strings.format("Unable to add: %s There's another index id with the same name", indexId);
```

Comment:
Nit: Since we already throw assertion error inside `noDuplicateIndicesWithSameName`, the assertion message here is no longer useful and can be removed.